src/disk_interface_test.cc

```diff
@@ -171,7 +171,7 @@ TEST_F(DiskInterfaceTest, StatCache) {
   EXPECT_GT(disk_.Stat("subdir/subsubdir", &err), 1);
   EXPECT_EQ("", err);
 
-#ifndef _MSC_VER // TODO: Investigate why. Also see https://github.com/ninja-build/ninja/pull/1423
+#if 0  // TODO: Investigate why. Also see https://github.com/ninja-build/ninja/pull/1423
```

Comment:
should be `#ifndef _WIN32` then, otherwise we would disable it for Linux, too.

Reply:
Ok, I'll do it, but note that there is already an #ifndef _WIN32 just before this test declaration, because this is (currently) the only platform with a StatCache.

Reply:
Ah I see! Then we probably shouldn't touch this line at all, should we? The test succeeded on AppVeyor for MinGW: https://ci.appveyor.com/project/nico/ninja/builds/47865014/job/eg5b0uic6jgrrida

Reply:
The test definitely fails when building with my Debian mingw installation, and running on a Windows 10 machine though, which is why I had to disable it there too. I have no idea why.

```
[3/398] MSVCHelperTest.NoReadOfStderrto stderr 
[207/398] DiskInterfaceTest.StatCache
*** Failure in /work2/github-digit/ninja/src/disk_interface_test.cc:175
disk_.Stat("subdir", &err) == disk_.Stat("subdir/.", &err)
*** Failure in /work2/github-digit/ninja/src/disk_interface_test.cc:178
disk_.Stat("subdir", &err) == disk_.Stat("subdir/subsubdir/..", &err)  
[398/398] BuildLogRecompactTest.Recompact
```

I suspect the issue is with the MSVCRT.DLL runtime being used, not the compiler that builds the code.

Reply:
I see. Still it runs fine in some cases and one of them is the CI. That's why I would like to keep it enabled.

Reply:
All right, we'll keep a local patch for our own CI then. I have removed the change to disk_interface_test.cc here.